### PR TITLE
19574 Consolidate fieldset styling

### PR DESCRIPTION
**Pull request by myabc**
Succeeds #2812
## What this PR/WP does:
- consolidates styling for various types of fieldsets: 
  - collapsable (also called toggable [sic], togglable fieldsets)
  - fieldsets with _Check all_ controls.
  - preview fieldsets
- hides the legend on simple filters (to sighted), giving a more similar appearance to advanced filters. 
- quite a bit of cleanup!
- applies missing fieldset (and some form) styling in various views in core.
- :warning: removes the default bold styling on links (this is something we will need to add back selectively in places).
## What this PR/WP does not do:
- apply missing form and fieldset styling in wiki menu item forms.
- apply missing fieldset styling in various views _in plugins_.

https://community.openproject.org/work_packages/19574

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://myabc-feature-19574-quick-win-fieldset-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://myabc-feature-19574-quick-win-fieldset-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by myabc** [comment]
@NobodysNightmare @floriank This is ready for your review.

## Designer feedback needed

IMO, the expand/collapse behaviour of collapsible fieldsets is not obvious enough:

seeded_project_-_edit_timeline_report_ganz_simple_-_openproject

**Comment by NobodysNightmare** [comment]
> removes the default bold styling on links

loving it already!

**Comment by NobodysNightmare** [comment]
@myabc Are you and @houndci okay with eachother? Not sure if you trust him in CSS questions, but he seems to be quite annoyed...

TBH: I also like empty lines after `}` ^^

**Comment by floriank** [comment]
I would not spend too much time on the old `timelines.scss` or timelines\* in general - did anyone summon @zzmedia to the party yet?
- according to the prioritization list, this is deprecated

**Comment by NobodysNightmare** [comment]
Not sure if caused by this PR, but the bottom most textbox is cut off:

bottom-cut

(assuming PR involvement because it is a `form--fieldset -collapsible`)

File: app/assets/stylesheets/content/_forms.sass [hunk 1]
```diff
@@ -187,6 +161,38 @@ fieldset.form--fieldset
...
+  margin-top:     -2.8rem
+  text-align:     right
+  color:          lighten($body-font-color, 10)
+  font-size:      1rem
+  font-style:     italic
+  line-height:    1.8
+
+.form--fieldset-control-container
+  padding: 0 0.25rem
+  background-color: white
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_forms.sass]: Is this intended? It looks awkward:

white-bg

Btw. the living Styleguide is great!
I can search for "fieldset" and look for all occurences of fieldsets that might have been broken by @myabc :)

File: app/views/boards/show.html.erb [hunk 2]
```diff
@@ -52,7 +52,7 @@ See doc/COPYRIGHT.rdoc for more details.
       <%= preview_link preview_board_topics_path(@board), 'message-form-preview', class: 'button' %>
       <%= link_to l(:button_cancel), "#", onclick: 'Element.hide("add-message")', class: 'button' %>
     <% end %>
-    <div id="preview" class="wiki"></div>
+    <div id="preview"></div>
```
Comment (NobodysNightmare) [inline app/views/boards/show.html.erb:55]: I am irritated that I find no `preview` _class_ here... But it works, so I just say nothing...

Comment (myabc) [inline app/views/boards/show.html.erb:55]: @NobodysNightmare yep, maybe the structure is a bit strange :unamused:  We don't need to apply any styling here though: the `<div>` container is only needed for inserting the Ajax-loaded preview.

Comment (NobodysNightmare) [inline app/views/boards/show.html.erb:55]: I see

**Comment by NobodysNightmare** [comment]
regarding "obviousness" of collabsibility:

At least in the context of the timelines edit screen it is quite obvious, because of the amount of collapsible fieldsets...

Until otherwise proven I would take it as good enough.

**Comment by myabc** [comment]
@NobodysNightmare 

> regarding "obviousness" of collabsibility:

thanks for the comment. I think it's worth getting feedback from a couple people, but probably good enough to be merged.

File: app/assets/stylesheets/content/_forms.sass (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/content/_forms.sass]: @NobodysNightmare fixed in 75cf992.

**Comment by NobodysNightmare** [comment]
Waiting if @zzmedia or @lindenthal add any feedback on the corresponding ticket...

Will be back at this PR later today ;)

**Comment by myabc** [comment]
## Designer feedback needed

I marked 5333cda as WIP, as the preview styling is pretty rudimentary. I don't think this is anything critical though.

seeded_project_-_add_news_-_openproject

File: app/views/boards/show.html.erb (hunk 2 above)
Comment (myabc) [inline app/views/boards/show.html.erb:55]: @NobodysNightmare I actually discussed doing further refactoring here with @floriank . See: 
- https://github.com/opf/openproject/pull/2502#issuecomment-73216414
- https://gist.github.com/myabc/7506ad8b27eb35a6a019

Maybe in a future release… :disappointed: 

We should also wait for some design feedback (see https://github.com/opf/openproject/pull/2815#issuecomment-92356117)

**Comment by myabc** [comment]
> I would not spend too much time on the old timelines.scss or timelines\* in general 

I hope much of the contents of `_timelines.scss` will go away once WP 19594 is completed (currently exists as PR #2810, but needs to be reissued).

> did anyone summon @zzmedia to the party yet?

also notified in the linked WP.

**Comment by floriank** [comment]
@myabc It was requested to move design discussion to the related epic - unfortunately in my opinion, but requested nonetheless.

**Comment by NobodysNightmare** [comment]
Will merge as feedback was positive and you announced to do further work in further PRs...

**Pull request merged**